src/parser/parse.ts

```diff
@@ -204,124 +199,159 @@ export class Parser<T extends ParserInput, TFlags extends OutputFlags<T['flags']
+    *  Value function can be resolved later.
+    */
+    const addValueFunction = (fws: FlagWithStrategy): FlagWithStrategy => {
+      const tokenLength = fws.tokens?.length
+      // user provided some input
+      if (tokenLength) {
+        // boolean
+        if (fws.inputFlag.flag.type === 'boolean' && fws.tokens?.at(-1)?.input) {
+          return {...fws, valueFunction: async (i: FlagWithStrategy) => parseFlagOrThrowError(i.tokens?.at(-1)?.input !== `--no-${i.inputFlag.name}`, i.inputFlag.flag, i.tokens?.at(-1), this.context)}
+        }
+
+        // multiple with custom delimiter
+        if (fws.inputFlag.flag.type === 'option' && fws.inputFlag.flag.delimiter && fws.inputFlag.flag.multiple) {
+          return {
+            ...fws, valueFunction: async (i: FlagWithStrategy) => (await Promise.all(
+              ((i.tokens ?? []).flatMap(token => (token.input as string).split(i.inputFlag.flag.delimiter as string)))
+              // trim, and remove surrounding doubleQuotes (which would hav been needed if the elements contain spaces)
+              .map(v => v.trim().replace(/^"(.*)"$/, '$1').replace(/^'(.*)'$/, '$1'))
+              .map(async v => parseFlagOrThrowError(v, i.inputFlag.flag, {...i.tokens?.at(-1) as FlagToken, input: v}, this.context)),
+            )).map(v => validateOptions(i.inputFlag.flag as OptionFlag<any>, v)),
           }
         }
+
+        // multiple in the oclif-core style
+        if (fws.inputFlag.flag.type === 'option' && fws.inputFlag.flag.multiple) {
+          return {...fws, valueFunction: async (i: FlagWithStrategy) => Promise.all((fws.tokens ?? []).map(token => parseFlagOrThrowError(validateOptions(i.inputFlag.flag as OptionFlag<any>, token.input as string), i.inputFlag.flag, token, this.context)))}
+        }
+
+        // simple option flag
+        if (fws.inputFlag.flag.type === 'option') {
+          return {...fws, valueFunction: async (i: FlagWithStrategy) => parseFlagOrThrowError(validateOptions(i.inputFlag.flag as OptionFlag<any>, fws.tokens?.at(-1)?.input as string), i.inputFlag.flag, fws.tokens?.at(-1), this.context)}
+        }
       }
-    }
 
-    for (const k of Object.keys(this.input.flags)) {
-      const flag = this.input.flags[k]
-      if (flags[k]) continue
-      if (flag.env && Reflect.has(process.env, flag.env)) {
-        const input = process.env[flag.env]
-        if (flag.type === 'option') {
-          if (input) {
-            this._validateOptions(flag, input)
-
-            flags[k] = await this._parseFlag(input, flag)
-          }
-        } else if (flag.type === 'boolean') {
-          // eslint-disable-next-line no-negated-condition
-          flags[k] = input !== undefined ? isTruthy(input) : false
+      // no input: env flags
+      if (fws.inputFlag.flag.env && process.env[fws.inputFlag.flag.env]) {
+        const valueFromEnv = process.env[fws.inputFlag.flag.env]
+        if (fws.inputFlag.flag.type === 'option' && typeof valueFromEnv === 'string') {
```

Comment:
NIT: `valueFromEnv` has type `string | undefined` so it could be simplified to just check if it exists